timeseries/src/autogluon/timeseries/models/statsmodels/models.py

```diff
@@ -0,0 +1,174 @@
+from typing import List
+
+import numpy as np
+import pandas as pd
+from statsmodels.tsa.exponential_smoothing.ets import ETSModel
+from statsmodels.tsa.statespace.sarimax import SARIMAX
+
+from autogluon.timeseries.dataset.ts_dataframe import TimeSeriesDataFrame
+from autogluon.timeseries.utils.seasonality import get_seasonality
+from autogluon.timeseries.utils.warning_filters import statsmodels_warning_filter
+
+from .abstract_statsmodels import AbstractStatsmodelsModel, FittedLocalModel
+
+
+class StatsmodelsETSModel(AbstractStatsmodelsModel):
+    """Exponential smoothing with trend and seasonality model.
+
+    Based on `statsmodels.tsa.exponential_smoothing.ets.ETSModel`.
+
+    See `AbstractStatsmodelsModel` for common parameters.
+
+
+    Other Parameters
+    ----------------
+    error : str, default = "add"
+        Error model. Allowed values are "add" and "mul".
+        Note that "mul" is only applicable for time series with positive values.
+    trend : str or None, default = "add"
+        Trend component model. Allowed values are "add", "mul" and None.
+        Note that "mul" is only applicable for time series with positive values.
+    damped_trend : bool, default = False
+        Whether or not the included trend component is damped.
+    seasonal : bool, default = True
+        Whether to enable additive seasonality.
+    seasonal_periods : int or None, default = None
+        Number of time steps in a complete seasonal cycle for seasonal models. For example, 7 for daily data with a
+        weekly cycle or 12 for monthly data with an annual cycle.
+        When set to None, seasonal_periods will be inferred from the frequency of the training data. Can also be
+        specified manually by providing an integer > 1.
+        If seasonal_periods (inferred or provided) is equal to 1, seasonality will be disabled.
+    maxiter : int, default = 1000
+        Number of iterations during optimization.
+    n_jobs : int, default = -1
+        Number of CPU cores used to fit the models in parallel. When set to -1, all CPU cores are used.
+    """
+
+    quantile_method_name = "pred_int"
+    statsmodels_allowed_init_args = [
+        "error",
+        "trend",
+        "damped_trend",
+        "seasonal",
+        "seasonal_period",
+    ]
+    statsmodels_allowed_fit_args = [
+        "maxiter",
+        "n_jobs",
+    ]
+
+    def _fit_local_model(
+        self, timeseries: pd.Series, default_model_init_args: dict, default_model_fit_args: dict
+    ) -> FittedLocalModel:
+        model_init_args = default_model_init_args.copy()
+        # Infer seasonal_periods if seasonal_periods is not given / is set to None
+        seasonal_periods = model_init_args.pop("seasonal_periods", None)
+        if seasonal_periods is None:
+            seasonal_periods = get_seasonality(self.freq)
+        # Disable seasonality if the model cannot be fit with seasonality
+        if len(timeseries) < 2 * seasonal_periods or seasonal_periods == 1:
```

Comment:
nit: this computation seems to be repeated for each thread. is there a way to already compute these once per `_fit` call?